### PR TITLE
add postprocessing_masks gpu version

### PR DESCRIPTION
**Pull request by xizi**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by AllentDan** [comment]
There should be a parameter in deploy_cfg that control the device to do the postprocess

File: mmdeploy/codebase/mmdet/deploy/object_detection_model.py [hunk 1]
```diff
@@ -215,7 +225,7 @@ def forward(self, img: Sequence[torch.Tensor], img_metas: Sequence[dict],
                 y_off = img_metas[i]['border'][0]
                 dets[:, [0, 2]] -= x_off
                 dets[:, [1, 3]] -= y_off
-                dets[:, :4] *= (dets[:, :4] > 0).astype(dets.dtype)
```
Comment (grimoire) [inline mmdeploy/codebase/mmdet/deploy/object_detection_model.py:218]: Why do you remove `astype`?

File: mmdeploy/codebase/mmdet/deploy/object_detection_model.py [hunk 2]
```diff
@@ -168,10 +178,8 @@ def postprocessing_masks(det_bboxes: np.ndarray,
                 grid[None, :, :, :],
                 align_corners=False)
 
-            mask = img_masks
-            mask = (mask >= mask_thr_binary).to(dtype=torch.bool)
```
Comment (grimoire) [inline mmdeploy/codebase/mmdet/deploy/object_detection_model.py:172]: Why do you remove this comparation?

File: mmdeploy/codebase/mmdet/deploy/object_detection_model.py (hunk 1 above)
Comment (xizi) [reply mmdeploy/codebase/mmdet/deploy/object_detection_model.py:218]: dets的数据类型是torch.tensor，dtype是float32，不需要astype

Comment (grimoire) [reply mmdeploy/codebase/mmdet/deploy/object_detection_model.py:218]: `(dets[:, :4] > 0)` will give you a bool tensor.

**Comment by grimoire** [comment]
Hi, please merge the latest master branch. We have done some changes that might fix the ci.

**Comment by grimoire** [comment]
Hi, as this PR https://github.com/open-mmlab/mmdeploy/pull/286 fix some ci error. Please merge the latest master branch.

File: mmdeploy/codebase/mmdet/deploy/object_detection_model.py (hunk 1 above)
Comment (xizi) [reply mmdeploy/codebase/mmdet/deploy/object_detection_model.py:218]: torch.tensor dtype float32与bool 相乘会自动转换为float32

Comment (grimoire) [reply mmdeploy/codebase/mmdet/deploy/object_detection_model.py:218]: I see. A better way is using `torch.nn.threshold`, which might be faster.
```python
import torch
from torch.profiler import profile, ProfilerActivity


def func_test(func,
              data,
              threshold,
              num_test=10,
              activate=ProfilerActivity.CPU):

    # warmup
    for _ in range(num_test):
        out = func(data, threshold)

    with profile(activities=[activate], record_shapes=True) as prof:
        for _ in range(num_test):
            out = func(data, threshold)

    print(prof.key_averages().table(
        sort_by="cpu_time_total", top_level_events_only=True, row_limit=30))

    return out


def main():
    data = torch.rand(1, 3, 224, 224)
    threshold = 0.5
    activate = ProfilerActivity.CPU

    out0 = func_test(
        lambda data, threshold: data * (data > threshold),
        data,
        threshold,
        activate=activate)
    out1 = func_test(
        lambda data, threshold: torch.nn.functional.threshold(
            data, threshold, 0),
        data,
        threshold,
        activate=activate)
    torch.testing.assert_allclose(out0, out1)


if __name__ == '__main__':
    main()

```

**Review by AllentDan — APPROVED** [review]

**Pull request merged**